### PR TITLE
HV-1290 Support exploring the type hierarchy for ValueExtractor definitions

### PR DESCRIPTION
**Pull request by gsmet**
* https://hibernate.atlassian.net/browse/HV-1290

### CODE REVIEW

**Comment by gsmet** [comment]
Rebased.

File: engine/src/test/java/org/hibernate/validator/test/internal/engine/valuehandling/ValueExtractorDefinitionInHierarchyTest.java [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+
+		public T getValue() {
+			return value;
+		}
+	}
+
+	private static class ValueExtractorDefinedInParentClassChild extends ValueExtractorDefinedInParentClassParent {
+
+		@Override
+		public void extractValues(Wrapper<@ExtractedValue ?> originalValue, ValueReceiver receiver) {
```
Comment (gunnarmorling) [inline engine/src/test/java/org/hibernate/validator/test/internal/engine/valuehandling/ValueExtractorDefinitionInHierarchyTest.java]: Why is it that `@ExtractedValue` shows up here (same for some other methods in this test)?

Comment (gsmet) [inline engine/src/test/java/org/hibernate/validator/test/internal/engine/valuehandling/ValueExtractorDefinitionInHierarchyTest.java]: I suppose some Eclipse generation magic used when I wrote the test. Not the first time I notice that. I'll fix it.

**Comment by gsmet** [comment]
Merged, thanks for the review.

**Pull request closed without merging**